### PR TITLE
add desired capabilities to sender and amqp short type

### PR DESCRIPTION
**Pull request by yunhaoling**
This PR is to add the missing desired capability and short type which is required by the EH idempotent feature: https://github.com/Azure/azure-sdk-for-python/issues/12427

### CODE REVIEW

File: uamqp/async_ops/client_async.py [hunk 1]
```diff
@@ -424,6 +424,11 @@ class SendClientAsync(client.SendClient, AMQPClientAsync):
      will assume successful receipt of the message and clear it from the queue. The
      default is `PeekLock`.
     :type receive_settle_mode: ~uamqp.constants.ReceiverSettleMode
+    :param desired_capabilities: The extension capabilities desired from the peer endpoint.
+     To create an desired_capabilities object, please do as follows:
```
Comment (swathipil) [inline uamqp/async_ops/client_async.py:428]:
```suggestion
     To create a desired_capabilities object, please do as follows:
```

File: uamqp/async_ops/sender_async.py [hunk 2]
```diff
@@ -55,6 +55,11 @@ class MessageSenderAsync(sender.MessageSender):
      will assume successful receipt of the message and clear it from the queue. The
      default is `PeekLock`.
     :type receive_settle_mode: ~uamqp.constants.ReceiverSettleMode
+    :param desired_capabilities: The extension capabilities desired from the peer endpoint.
+     To create an desired_capabilities object, please do as follows:
```
Comment (swathipil) [inline uamqp/async_ops/sender_async.py:59]:
```suggestion
     To create a desired_capabilities object, please do as follows:
```

File: uamqp/client.py [hunk 3]
```diff
@@ -439,6 +439,11 @@ class SendClient(AMQPClient):
      will assume successful receipt of the message and clear it from the queue. The
      default is `PeekLock`.
     :type receive_settle_mode: ~uamqp.constants.ReceiverSettleMode
+    :param desired_capabilities: The extension capabilities desired from the peer endpoint.
+     To create an desired_capabilities object, please do as follows:
```
Comment (swathipil) [inline uamqp/client.py:443]:
```suggestion
     To create a desired_capabilities object, please do as follows:
```

File: uamqp/sender.py [hunk 4]
```diff
@@ -50,6 +50,11 @@ class MessageSender(object):
      will assume successful receipt of the message and clear it from the queue. The
      default is `PeekLock`.
     :type receive_settle_mode: ~uamqp.constants.ReceiverSettleMode
+    :param desired_capabilities: The extension capabilities desired from the peer endpoint.
+     To create an desired_capabilities object, please do as follows:
```
Comment (swathipil) [inline uamqp/sender.py:54]:
```suggestion
     To create a desired_capabilities object, please do as follows:
```

**Review by swathipil — COMMENTED** [review]
extra small nits XD

File: uamqp/async_ops/client_async.py (hunk 1 above)
Comment (yunhaoling) [reply uamqp/async_ops/client_async.py]: thanks!

File: uamqp/async_ops/client_async.py [hunk 5]
```diff
@@ -424,6 +424,11 @@ class SendClientAsync(client.SendClient, AMQPClientAsync):
      will assume successful receipt of the message and clear it from the queue. The
      default is `PeekLock`.
     :type receive_settle_mode: ~uamqp.constants.ReceiverSettleMode
+    :param desired_capabilities: The extension capabilities desired from the peer endpoint.
+     To create a desired_capabilities object, please do as follows:
+        - 1. Create an array of desired capability symbols: `capabilities_symbol_array = [types.AMQPSymbol(string)]`
+        - 2. Transform the array to AMQPValue object: `utils.data_factory(types.AMQPArray(capabilities_symbol_array))`
```
Comment (annatisch) [inline uamqp/async_ops/client_async.py:430]: Not blocking right now - but could we not automatically perform this transformation if someone passes in a list of strings?
So type could be Union[AMQPValue, List[str]]

**Review by annatisch — APPROVED** [review]
LGTM

File: uamqp/async_ops/client_async.py (hunk 5 above)
Comment (yunhaoling) [reply uamqp/async_ops/client_async.py:430]: yea, I think that should be doable and would be nice to have.

I have created an separate issue for this one: https://github.com/Azure/azure-uamqp-python/issues/212

**Pull request merged**